### PR TITLE
DataFrame.display injection

### LINKED ISSUE #513: Add `.display` extension function that allows specifying the number of rows
See 

https://github.com/locationtech/rasterframes/blob/8bbed00ab7970a4003533cadc99173e2a5ab7838/pyrasterframes/src/main/python/pyrasterframes/rf_ipython.py#L263

### PR DESCRIPTION
**Pull request by metasim**
* Patched in `pyspark.sql.DataFrame.display` function for passing parameters to html rendering code normally invoked through display hooks.
* Updated Python dependencies.
* Set default `prefer-gdal` to `false` as JVM parser is just as good for most use cases.

Closes #513 

TODO:

* [x] add to users' manual 
* [x] fix build issue with `branca`:
  `NotImplementedError: Can't perform this operation for unregistered loader type`

### CODE REVIEW

File: core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 rasterframes {
   nominal-tile-size = 256
-  prefer-gdal = true
+  prefer-gdal = false
```
Comment (vpipkt) [inline core/src/main/resources/reference.conf:3]: i like it. will simplify a lot of use cases particularly for folks new to the project.

File: pyrasterframes/src/main/python/README.md [hunk 2]
```diff
@@ -38,6 +38,15 @@ Issue tracking is through [github](https://github.com/locationtech/rasterframes/
 
 Community contributions are always welcome. To get started, please review our [contribution guidelines](https://github.com/locationtech/rasterframes/blob/develop/CONTRIBUTING.md), [code of conduct](https://github.com/locationtech/rasterframes/blob/develop/CODE_OF_CONDUCT.md), and [developer's guide](../../../README.md).  Reach out to us on [gitter][gitter] so the community can help you get started!
 
+## Development environment setup
```
Comment (vpipkt) [inline pyrasterframes/src/main/python/README.md:41]: nice.

File: pyrasterframes/src/main/python/tests/IpythonTests.py [hunk 3]
```diff
@@ -60,3 +63,23 @@ def test_all_nodata_tile(self):
...
+        import pyrasterframes.rf_ipython
+
+        self.create_layer()
+        ip = globalipapp.get_ipython()
+
+        num_rows = 2
+        row_count = 0
+
+        def counter(data, _):
+            nonlocal row_count
```
Comment (vpipkt) [inline pyrasterframes/src/main/python/tests/IpythonTests.py:78]: 🤔

**Review by vpipkt — COMMENTED** [review]
Looking good so far!

File: pyrasterframes/src/main/python/tests/IpythonTests.py (hunk 3 above)
Comment (metasim) [reply pyrasterframes/src/main/python/tests/IpythonTests.py:78]: New in Python 3.x, so you don't have to use `global` and have the variable in the outermost scope.

File: pyrasterframes/src/main/python/docs/ipython.pymd [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# IPython/Jupyter Extensions
```
Comment (vpipkt) [inline pyrasterframes/src/main/python/docs/ipython.pymd:1]: i think you need this?

```suggestion
```python, setup, echo=False
from IPython.core.display import display
from pyrasterframes.utils import create_rf_spark_session
from pyrasterframes.rasterfunctions import *

spark = create_rf_spark_session()
```

# IPython/Jupyter Extensions
```

File: pyrasterframes/src/main/python/docs/ipython.pymd [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ 
+Print the schema to confirm its "shape":
+
+```python schema
+df.printSchema()
+```
+
+# Tile Display
+
+Let's look at a single tile. A `pyrasterframes.rf_types.Tile` will automatically render nicely in Jupyter or IPython.
```
Comment (vpipkt) [inline pyrasterframes/src/main/python/docs/ipython.pymd:37]: i think some of this is already covered in `raster-write.pymd`, would be nice to slim that page down and point it over here. you may also want to just lift some of that content into here.

**Review by vpipkt — CHANGES_REQUESTED** [review]
i couldn't get the docs to build to really see whats goin on but looks pretty good. a few small changes

**Comment by metasim** [comment]
> i couldn't get the docs to build to really see whats goin on but looks pretty good. a few small changes

Hint on what was going wrong?

**Comment by vpipkt** [comment]
> > i couldn't get the docs to build to really see whats goin on but looks pretty good. a few small changes
> 
> Hint on what was going wrong?

just some environment cruft on my end.

**Comment by metasim** [comment]
@vpipkt FYI, there seems to be a regression in the _Unsupervised Machine Learning_ section of the documentation, where the overview prediction raster rendering is a single color. Will deal with in a subsequent PR.

**Pull request merged**